### ISSUE TITLE
[Linux] SDL2-based motion controls not recognized

### ISSUE THREAD

**Issue by Shoegzer** [yuzu-emu/yuzu#6325]
Tested this new feature against commit bfe8816, by selecting "Pro Controller" and "PS4 Controller 0" as input device, then clicking "Motion 1" in the UI. The message "Shake!" remains even after shaking the controller, and nothing appears to be set.

To rule out this being hidraw permissions related, I added the following udev rule in `/etc/udev/rules.d/99-hidraw.rules`:

`KERNEL=="hidraw*", SUBSYSTEM=="hidraw", MODE="0664", GROUP="plugdev"`

I then restarted with `udevadm control --reload` and ran the test again but still no motion was detected.  I also verified that the user running yuzu is in the plugdev group.

Test platform is Linux Mint 20.1 / kernel 5.11 with ds4 controller connected via bluetooth. All other controller settings (button mappings, rumble etc.) work properly. Tested against a yuzu binary compiled with gcc 10.2.0 straight from the source directory build/bin/yuzu.

**Comment by german77** [comment]
After #6244 if motion it detected it will be automapped. If this doesn't happen SDL failed to enable motion for that controller. This is probably caused if SDL can't use the hidapi driver

**Comment by v1993** [comment]
You've confirmed that you are in `plugdev` group. Assuming you got it right, your last bet is to check that hidraw *headers* are installed and install them if they aren't. Then purge yuzu's build dir and build again, taking note of `HIDAPI:   ON/OFF (wanted: ON)` line during CMake step.

**Comment by Shoegzer** [comment]
Thanks @v1993.  I've confirmed installation of `libhidapi-dev` (which includes `/usr/include/hidapi/hidapi.h`).  However, CMake shows:

`--   HIDAPI                 (Wanted: ON): OFF`

[Build log](https://pastebin.com/F0SDgVuR)
[CMakeOutput.log](https://pastebin.com/QjiDvbHv)

**Comment by v1993** [comment]
Have you purged build directory first? Run `rm -rf *` in it.

**Comment by Shoegzer** [comment]
Yes, I've built from a clean directory in testing.

**Comment by v1993** [comment]
@Shoegzer Can you, please, try latest yuzu appimage build and see if it works?

**Comment by Shoegzer** [comment]
Sure, motion works with the ds4 controller on the appimage build - Motion 1 is set to "SDL Motion" and the feature is properly detected ingame.

This makes me wonder further why hidapi is not detected on compiled builds - is it due to another missing dependency or maybe the SDL2 external?

**Comment by v1993** [comment]
SDL2 is specifically set to build with hidapi support when available, which works both on CI and testing machines and appimage works for you. Given this I come to conclusion that it's some problem with your installation of hidapi, which isn't really an issue with yuzu.

On the other hand, your case may signify that there should be more explicit check for hidapi being used by SDL to avoid issues like yours.

**Comment by Shoegzer** [comment]
Interesting, thanks for the explanation.  The Mint 20.1 test system isn't much different than stock though with an apt install of `libhidapi-dev`, odd that it's not detected.  What is it looking for specifically?  I assume that by CMake showing hidapi as off, it's not finding `/usr/include/hidapi/hidapi.h`.. or is it looking for some other dependency that hasn't been installed?  An explicit check does seem like it would help.

By the way, here's hidapi.h on the system, let me know if you think further verification info like this might help:
```
$ ls -la /usr/include/hidapi/
total 36
drwxr-xr-x   2 root root  4096 May 17 11:24 .
drwxr-xr-x 164 root root 16384 May 17 11:24 ..
-rw-r--r--   1 root root 13898 Jul 11  2019 hidapi.h
```

**Comment by v1993** [comment]
You can try installing libusb development packages. I'm not sure what else to suggest.

SDL uses some rather weird check for hidapi presence which I didn't explored too much.

**Comment by Shoegzer** [comment]
@v1993 bingo, that was it!  I installed `libusb-1.0-0-dev` from the ubuntu repo and CMake now detects hidapi.  In testing the compiled binary, motion is  detected in both the UI and ingame.  Also, as a test I removed the udev hidraw rule and reloaded the rules, and motion was still detected and working properly, so apparently that's not necessary in setups like this one either.

Testing further, I did detect a strange issue with Super Mario Odyssey where motion works even if it's turned off in the game menu.  I'm not sure if that's a game or emulation bug, and I can open a separate issue for it but I'll defer to you on that one.

I'll close this issue now. Thank you for your assistance and information!  I hope my info helped you too.
